### PR TITLE
Fix handling of scale transforms when determing view coordinates

### PR DESCRIPTION
**Pull request by mmaatttt**
We encountered an element targeting bug on a call where the left (x) and top (y) positions of all the elements was 3x greater than it should be. However the size of the target was correct.

The problematic code: https://github.com/appcues/appcues-flutter-plugin/blob/47ac3ab137a160d0b067fec7cf0c1eb56762903b/lib/appcues_flutter.dart#L358-L377

Of note:
1. Only translate transforms are handled (L361 or fallback to L369). Scale operations are ignored. So for the simple case, any widget wrapped with `Transform.scale(scale: 2, child: <widget>)` would not have a properly calculated target.
2. `MatrixUtils.getAsScale` (L367) returns non-null "if the matrix is nothing but a symmetric 2D scale transform" ([doc](https://api.flutter.dev/flutter/painting/MatrixUtils/getAsScale.html)). If a transform matrix includes a scale AND translate, this will return `null`.
    - In this `null` case (L369-375), the top left of the rectangle is transformed, but the size of the element (ie right and bottom) are not similarly transformed. This matches the scenario that raised this issue where the left and top of the elements were wrong but the size was right :bulb:
    - The reason the issue wasn't present in our example app and other customers is that the root scale transform (for the device pixel density) was a pure scale transform and so failed the check on L367 and no transform was applied.

> [!IMPORTANT]
> **TL;DR:** we need to handle scale transforms.

## Solution

Always apply the transform without worrying about whether it's a translate or scale. This means all the scenarios are properly handled without the need for fallbacks like we introduced in #76 (I've specifically tested that scenario with this new approach and it's working properly). Always applying the transform means the final rectangle we get is scaled up by the `devicePixelRatio`. Therefore we need to scale the rectangle back down to logical pixels.

I also optimized a bit by moving `transformToRoot` to be a static function instead of defining it inside a condition in the function that gets called on each node. I don't know the specific details of how Dart would optimize it, but intuitively it should be better now.

## Testing

||Regular widget|Scaled widget|FAB||
|-|-|-|-|-|
|3x screen iPhone|✅|✅|✅|Simulator Screenshot - iPhone 15 Pro - 2024-12-06 at 10 43 34|
|2x screen iPhone|✅|✅|✅|Simulator Screenshot - iPhone 8 - 2024-12-06 at 11 10 42|
| 2.625x screen Android |✅|✅|✅|Screenshot_20241206_110910|

### CODE REVIEW

**Review by andretortolano — APPROVED** [review]
Healthy change to the sdk, all gradle files changes are related to the gradle update and dart code updates the visuals (previous broken) of the sample app + the required fix.

File: example/lib/src/screens/events.dart [hunk 1]
```diff
@@ -15,7 +15,7 @@ class EventsScreen extends StatelessWidget {
             onPressed: () {
               Appcues.debug();
             },
-            child: const Text('Debug', style: TextStyle(color: Colors.white)),
+            child: const Text('Debug'),
```
Comment (iujames) [inline example/lib/src/screens/events.dart:18]: ah thanks for fixing this, i noticed that one once before

**Review by iujames — APPROVED** [review]
great find and update, this seems a much more robust yet simpler solution than before 🎉
